php_phongo.c

```diff
@@ -852,8 +852,15 @@ int phongo_execute_command(mongoc_client_t* client, php_phongo_command_type_t ty
 	if (!result) {
 		if (error.domain == MONGOC_ERROR_SERVER || error.domain == MONGOC_ERROR_WRITE_CONCERN) {
 			zend_throw_exception(php_phongo_commandexception_ce, error.message, error.code TSRMLS_CC);
-			php_phongo_bson_to_zval(bson_get_data(&reply), reply.len, return_value);
-			phongo_add_exception_prop(ZEND_STRL("resultDocument"), return_value TSRMLS_CC);
+#if PHP_VERSION_ID >= 70000
+			zval zv;
```

Comment:
Declarations need to appear first in a scope. You may need to break this up as:

```c
#if PHP_VERSION_ID >= 70000
zval zv;
#else
zval *zv;
#endif

zend_throw_exception(php_phongo_commandexception_ce, error.message, error.code TSRMLS_CC);
php_phongo_bson_to_zval(bson_get_data(&reply), reply.len, &zv);

#if PHP_VERSION_ID >= 70000
phongo_add_exception_prop(ZEND_STRL("resultDocument"), &zv);
#else
phongo_add_exception_prop(ZEND_STRL("resultDocument"), zv TSRMLS_CC);
#endif
```

I omitted `TSRMLS_CC` for PHP 7 since it's defined as nothing (regardless of whether PHP 7 is compiled with thread safety). This is only need for PHP 5, so we can omit it in any PHP 7-specific code.